backend/lucene/src/main/java/org/hibernate/search/backend/lucene/types/predicate/impl/LuceneFieldPredicateBuilderFactory.java

```diff
@@ -32,6 +33,9 @@
 	 */
 	boolean isDslCompatibleWith(LuceneFieldPredicateBuilderFactory other);
 
+	MatchIdPredicateBuilder<LuceneSearchPredicateBuilder> createMatchIdPredicateBuilder(
```

Comment:
As for the Elasticsearch backend, this is unnecessary; you can remove this method and its implementations.

Reply:
It seems you didn't take my comment into account? This method and its implementations should be removed.